src/commands/ingress/tryConfigureIngressUsingDockerfile.ts

```diff
@@ -0,0 +1,100 @@
+*--------------------------------------------------------------------------------------------*/
+
+import { AzExtFsExtra } from '@microsoft/vscode-azext-utils';
+import { ext } from '../../extensionVariables';
+import { localize } from '../../utils/localize';
+import type { IngressContext } from './IngressContext';
+import { getDefaultPort } from './editTargetPort/getDefaultPort';
+
+export async function tryConfigureIngressUsingDockerfile(context: IngressContext): Promise<void> {
+    if (!context.dockerfilePath) {
+        return;
+    }
+
+    context.dockerfileExposePorts = await getDockerfileExposePort(context.dockerfilePath);
+
+    if (context.alwaysPromptIngress) {
+        return;
+    }
+
+    if (!context.dockerfileExposePorts) {
+        context.enableIngress = false;
+        context.enableExternal = false;
+    } else if (context.dockerfileExposePorts) {
+        context.enableIngress = true;
+        context.enableExternal = true;
+        context.targetPort = getDefaultPort(context);
+    }
+
+    // If a container app already exists, activity children will be added automatically in later execute steps
+    // if (!context.containerApp) {
+    //     context.activityChildren?.push(
+    //         new GenericTreeItem(undefined, {
+    //             contextValue: createActivityChildContext(['ingressPromptStep', activitySuccessContext]),
+    //             label: context.enableIngress ?
+    //                 localize('ingressEnableLabel', 'Enable ingress on port {0} (found Dockerfile configuration)', context.targetPort) :
+    //                 localize('ingressDisableLabel', 'Disable ingress (found Dockerfile configuration)'),
+    //             iconPath: activitySuccessIcon
+    //         })
+    //     );
+    // }
+
+    ext.outputChannel.appendLog(context.enableIngress ?
+        localize('ingressEnabledLabel', 'Detected ingress on port {0} using Dockerfile configuration.', context.targetPort) :
+        localize('ingressDisabledLabel', 'Detected no ingress using Dockerfile configuration.')
+    );
+}
+
+export class PortRange {
+    private readonly _start: number;
+    private readonly _end: number;
+
+    constructor(start: number, end?: number) {
+        this._start = start;
+        this._end = end ? end : start;
+    }
+
+    get start(): number {
+        return this._start;
+    }
+
+    get end(): number {
+        return this._end;
+    }
+
+    includes(port: number): boolean {
+        return port >= this.start && port <= this.end;
+    }
+}
+
+export async function getDockerfileExposePort(dockerfilePath: string): Promise<PortRange[] | undefined> {
+    if (!await AzExtFsExtra.pathExists(dockerfilePath)) {
+        return undefined;
+    }
+
+    const content: string = await AzExtFsExtra.readFile(dockerfilePath);
+    const lines: string[] = content.split('\n');
+
+    const portRanges: PortRange[] = [];
+    for (const line of lines) {
+        if (/^EXPOSE/i.test(line.trim())) {
```

Comment:
Dockerfiles aren't too tough to parse, but it may be easier to use an existing parser. The Dockerfile language service used by the Docker extension has one. https://www.npmjs.com/package/dockerfile-ast

Reply:
Hmm, I did look into this one before and I ended up feeling unsure if it removed enough complexity to be worth it since my use case is relatively small here.

(1) I still have to loop through each line similar to what I'm already doing and search for an `EXPOSE` keyword 

(2) The args picked up are still raw strings that I have to manually loop through to identify ranges `-` and slashes `/` so that I can convert them into useable port numbers.

I think the one main benefit is that it will reduce the complexity of the `RegExp` that I'm using, but the overall process doesn't seem to get that much simpler.  Do you feel it may still be beneficial to use that dep?

Reply:
That's fair, I think it's fine to manually parse, just make sure you know all the [variations](https://docs.docker.com/engine/reference/builder/#expose) that an EXPOSE statement can have.